flink-dstl/flink-dstl-dfs/src/test/java/org/apache/flink/changelog/fs/FsStateChangelogWriterTest.java

```diff
@@ -246,15 +248,233 @@ void testFileAvailableAfterClose() throws Exception {
         }
     }
 
+    @Test
+    void testLocalFileDiscard() throws Exception {
```

Comment:
I tried to revert the code changes in this PR, this unit test could still pass. It seems the added test is useless.

Reply:
Thanks for the comments, as @zoltar9264 mentioned, this UT is used to verify that the newly changed code can correctly delete the local file. It doesn't have a bad effect on current code, I tend to keep it here.

Reply:
If so, why we cannot introduce a UT would fail with code before this PR?

Reply:
@Myasuka 
`testLocalFileAfterMaterialize()` would fail with code before this PR,  because the previous test of the local dstl files are not enough, `testLocalFileDiscard()` is a supplement.